### DISCUSSION TITLE
Login/Authorization System Migration to uWebSockets.js: Cookie Reading Issue

### DISCUSSION THREAD

**Discussion by BroderTheCuriousDev** [uNetworking/uWebSockets.js#985] [Q&A] [answered]
### **Background**
I am trying to build a login/authorization system on Uwebsocket. I recently migrated from socket.io and 
came to the conclusion that I can not use passport.js anymore. I decided to rewrite the authorization system from scratch. 

### **Challenge**
The problem I am encountering is the cookie I set in a .get route can not be read in the ws.upgrade handler.
When trying to use the req.getHeader('cookie') it returns nothing. 

I have tried just about everything and read through all the discussions and issues I can find on this Repo but to no avail.

### **Reference**

I stumbled across this answer on a different discussion [How to implement user authorization? #112](https://github.com/uNetworking/uWebSockets.js/discussions/112#discussioncomment-177625) but this code yields no better results. 

### **Objective** 
1. Authenticate the user in an HTTP route. Generate JWT token
2. Set the cookie on the user
3. Read the cookie in the WS upgrade handler

### **Code Example**

```
const {App} = require('uWebSockets.js')

  const app = App()
  app
    .ws('/*', {
      upgrade(res, req, context) {

          const cookies = req.getHeader('cookie') 
          // Implement cookie validation and other logic
  
          console.log(cookies)
  
          res.upgrade(
          {},
          req.getHeader('sec-websocket-key'),
          req.getHeader('sec-websocket-protocol'),
          req.getHeader('sec-websocket-extensions'),
          context
          )
      },
      open(ws) { }
    })
    .any('/*', (res, req) => {
      // Set the cookie 
      res.writeHeader('Set-Cookie', '_token=jwt; SameSite=Strict; HttpOnly') // Only example
      res.end('cookie sample')
    })
    .listen('::', 9001, token => {
      console.info(token)
    })
```

Any insights or suggestions on overcoming this cookie reading issue would be highly appreciated. Thank you!

**Comment by e3dio** [comment, marked as answer]
You are lacking basic troubleshooting steps. Did you check browser to confirm cookie was set, and was sent with request via chrome dev tools ?

**Reply by e3dio** [reply]
Your exact example code works fine, did you try your example or you copy/pasted select parts of your app and did not try it ?

**Reply by BroderTheCuriousDev** [reply]
Hey. I did in fact try it. After getting your confirmation that the code does indeed work I started to look at external factors. I was using live-server to host my temporary index.html file which opened a server at 127.0.0.1 instead of localhost. This is something I missed in the heat of the moment. I thought I was missing something related to Uwebsocket.js. Thank you for pointing me in the right direction!
